### PR TITLE
Add additonal Stm32g0 socs and nucleog0b1 board.

### PR DESCRIPTION
**Pull request by str4t0m**
- dts/arm/st: g0: remove unnecessary interrupt-names from adc, and uart
- dts/arm/st: add additional g0 device definitions
- soc/arm/st_stm32: Add additional stm32g0 socs
- drivers: gpio: add definition for stm32g0 porte
- drivers: clock_control: stm32g0: add support for g051, g0b1, g0b0 socs
- board: arm: add support for the nucleo_g0b1re board
- drivers: clock_control: stm32: rm Kconfig for APB2 prescaler on g0 series
- boards: stm32g0: add jlink and stm32cubeprogrammer runners.

Flashing of stm32g0b1 currently works with stm32cubeprogrammer and jlink.
Keil has already released a dsp for the new series, but some issues lead to pyocd flashing failure,
therefore i have added stm32g071 as target for pyocd for now.

### CODE REVIEW

File: dts/arm/st/g0/stm32g070X6.dtsi
Comment (erwango) [inline dts/arm/st/g0/stm32g070X6.dtsi:1]: I haven't seen any reference to this one (only STM32G070CB STM32G070KB STM32070RB)
Do you have any link ?

File: dts/arm/st/g0/stm32g070X8.dtsi
Comment (erwango) [inline dts/arm/st/g0/stm32g070X8.dtsi:1]: Same here

File: dts/arm/st/g0/stm32g071X6.dtsi
Comment (erwango) [inline dts/arm/st/g0/stm32g071X6.dtsi:1]: This one is not known neither

File: dts/arm/st/g0/stm32g0b0.dtsi
Comment (erwango) [inline dts/arm/st/g0/stm32g0b0.dtsi:81]: We're not using interrupt names on spi nodes

File: dts/arm/st/g0/stm32g0b1.dtsi
Comment (erwango) [inline dts/arm/st/g0/stm32g0b1.dtsi:27]: Please remove interrupt-names

File: dts/arm/st/g0/stm32g0b1.dtsi
Comment (erwango) [inline dts/arm/st/g0/stm32g0b1.dtsi:36]: Please remove interrupt-names

File: dts/arm/st/g0/stm32g0b1.dtsi
Comment (erwango) [inline dts/arm/st/g0/stm32g0b1.dtsi:47]: Please remove interrupt-names

File: dts/arm/st/g0/stm32g0b1.dtsi
Comment (erwango) [inline dts/arm/st/g0/stm32g0b1.dtsi:91]: Please remove interrupt-names

File: dts/arm/st/g0/stm32g0b1Xb.dtsi
Comment (erwango) [inline dts/arm/st/g0/stm32g0b1Xb.dtsi:12]: 36

**Review by erwango — CHANGES_REQUESTED** [review]

**Comment by erwango** [comment]
@str4t0m Thanks for the work! Some changes required

File: dts/arm/st/g0/stm32g0.dtsi
Comment (str4t0m) [reply dts/arm/st/g0/stm32g0.dtsi]: is now removed.

File: dts/arm/st/g0/stm32g0.dtsi
Comment (str4t0m) [reply dts/arm/st/g0/stm32g0.dtsi]: I have left it "global", as there is currently no benefit in changing it.

File: dts/arm/st/g0/stm32g0.dtsi
Comment (str4t0m) [reply dts/arm/st/g0/stm32g0.dtsi:194]: This has now been moved into pr #32646

File: dts/arm/st/g0/stm32g031.dtsi
Comment (str4t0m) [reply dts/arm/st/g0/stm32g031.dtsi]: done

File: dts/arm/st/g0/stm32g070X6.dtsi
Comment (str4t0m) [reply dts/arm/st/g0/stm32g070X6.dtsi]: I have accidentally duplicated that from 71x6.

File: dts/arm/st/g0/stm32g070X8.dtsi
Comment (str4t0m) [reply dts/arm/st/g0/stm32g070X8.dtsi]: removed

File: dts/arm/st/g0/stm32g071X6.dtsi
Comment (str4t0m) [reply dts/arm/st/g0/stm32g071X6.dtsi:1]: It is listed in the Keil pack, but that could be an error. Should I remove it?

File: dts/arm/st/g0/stm32g0b1Xb.dtsi
Comment (str4t0m) [reply dts/arm/st/g0/stm32g0b1Xb.dtsi:12]: The 144k should be right.

**Comment by str4t0m** [comment]
I have seen that the STM32_open_pin_data repo has been updated today (stm32cubemx 6.2.0 release).
- This is the first one that includes the definitions of the g05x socs.
- And for g0b1 it now includes the 128k flash variant:
  Additionally to the mcu/STM32G0B1R(C-E)Tx.xml they have now added a mcu/STM32G0B1R(B-C-E)Tx.xml  file...

Will we need to wait until the the pinctrl DTS files of the new release are merged in the hal before this pr can be merged, or could this pr be merged before?

File: dts/arm/st/g0/stm32g071X6.dtsi
Comment (str4t0m) [reply dts/arm/st/g0/stm32g071X6.dtsi]: I have removed it now as I only found this in the keil dfp and not on the ST website.

**Comment by str4t0m** [comment]
I addressed all changes and rebased, @erwango could you have another look?

**Review by erwango — APPROVED** [review]

**Review by FRASTM — APPROVED** [review]

**Pull request merged**
